hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/TestHddsDatanodeService.java

```diff
@@ -155,20 +154,23 @@ public void testTmpDirOnShutdown(String schemaVersion) throws IOException {
         clusterId, conf, LOG, null);
     // Create a container and move it under the tmp delete dir.
     KeyValueContainer container = ContainerTestUtils
-        .setUpTestContainerUnderTmpDir(
+        .addContainerToDeletedDir(
             hddsVolume, clusterId, conf, schemaVersion);
     assertTrue(container.getContainerFile().exists());
     assertTrue(container.getContainerDBFile().exists());
+    File[] deletedContainersAfterShutdown =
+        hddsVolume.getDeletedContainerDir().listFiles();
+    assertNotNull(deletedContainersAfterShutdown);
+    assertEquals(1, deletedContainersAfterShutdown.length);
 
     service.stop();
     service.join();
     service.close();
 
-    ListIterator<File> deleteLeftoverIt = KeyValueContainerUtil
-        .ContainerDeleteDirectory.getDeleteLeftovers(hddsVolume);
-    assertFalse(deleteLeftoverIt.hasNext());
-
-    volumeSet.shutdown();
+    deletedContainersAfterShutdown =
```

Comment:
From code changes, unable to get when cleanup is triggered? as during HddsVolume shutdown, cleanup is removed.

Reply:
`HddsVolume#shutdown` calls `HddsVolume#cleanDeletedContainerDir` to do clear out the directory on shutdown.

`HddsDatanodeService#stop -> DatanodeStateMachine#stopDaemon -> DatanodeStateMachine#close -> OzoneContainer#stop -> MutableVolumeSet#shutdown -> StorageVolume#shutdown`. The last call is overridden if the `StorageVolume` is an `HddsVolume`.